src/web/pool.rs

```diff
@@ -4,14 +4,24 @@ use postgres::Connection;
 use r2d2;
 use r2d2_postgres;
 
+#[cfg(test)]
+use std::sync::{Arc, Mutex, MutexGuard};
+
 pub(crate) enum Pool {
     R2D2(r2d2::Pool<r2d2_postgres::PostgresConnectionManager>),
+    #[cfg(test)]
+    Simple(Arc<Mutex<Connection>>),
```

Comment:
Instead of making this an enum where we only use the `Simple` variant in test mode, would it be easier to make the whole enum a cfg() parameter?

```rust
#[cfg(not(test))]
type Pool = r2d2::Pool<r2d2_postgres::PostgresConnectionManager>;
#[cfg(test)]
type Pool = Arc<Mutex<Connection>>;
```

That would also prevent accidentally modifying the prod database during tests (not that we plan to run tests in prod,  but it would also avoid screwing up my dev environment :P)

Reply:
It would be really painful to do this code wise: during `#[cfg(test)]` all the code is compiled, not just tests, and the rest of the application still expects a connection pool (for example when starting the web server from the binary).